src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl_ModelComputation.cs

```diff
@@ -192,12 +208,18 @@ private async ValueTask HighlightExpandAndPresentItemsAsync(ImmutableSegmentedLi
             }
             else
             {
-                // On Document Outline Control initialization, SymbolTree.ItemsSource is null
-                if (SymbolTree.ItemsSource is not null)
+                if (SymbolTreeIsInitialized)
                     DocumentOutlineHelper.UnselectAll((IEnumerable<DocumentSymbolUIItem>)SymbolTree.ItemsSource);
             }
 
             SymbolTree.ItemsSource = documentSymbolUIItems;
+
+            // If we are setting SymbolTree.ItemsSource for the first time, hide the progress bar.
+            if (!SymbolTreeIsInitialized)
```

Comment:
this check doesn't seem necessary, right.  you can always just call UpdateProgressBarVisibility(Visibility.Hidden);.  And that code can check if the visbility would change and not do anything if it wouldn't.